managed/services/user/user.go

```diff
@@ -85,8 +94,10 @@ func (s *Service) UpdateUser(ctx context.Context, req *userpb.UserUpdateRequest)
 		return nil, status.Errorf(codes.InvalidArgument, "Tour flag cannot be unset")
 	}
 
-	// TODO : Get ID from Grafana
-	userID := 32
+	userID, err := s.c.GetUserID(ctx)
+	if err != nil {
+		return nil, err
+	}
 
 	userInfo := &models.UserDetails{}
 	e := s.db.InTransaction(func(tx *reform.TX) error {
```

Comment:
let's merge these two transactions

Reply:
In addition to the @BupycHuk suggestion - as far as I remember, even if the SELECT and then the UPDATE queries being called inside a single transaction it doesn't guarantee that the selected rows not to be changed between those calls. The SELECT FOR UPDATE does.

Reply:
@artemgavrilov have we fixed that?

Reply:
@BupycHuk Well, technically it's not a bug or something that should be fixed. We have a ticket about increasing default transaction isolation level https://jira.percona.com/browse/PMM-8433 up to `serializable`. After that there will be no need to use `select for update`.

However in this particular case we can use simple `select` because update didn't use anything from selected data. `select for update` needed when you have logic like this:

```
SELECT money FROM user WHERE id = 1 FOR UPDATE;
var newMoney = money + 100
UPDATE user SET money = newMoney WHERE id = 1;
```

Moreover I don't know why we need `select` here at all, we can just use `update`. DB will retrun error if user is not exists.

Reply:
Merged transactions. Skipping SELECT FOR usage for now. Let me know if it's compulsory.